console/console-init/ui/src/modules/project-detail/components/AccessCredentials/AccessCredentials.tsx

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import {
+  Card,
+  CardBody,
+  CardHeader,
+  PageSection,
+  Split,
+  SplitItem,
+  Text,
+  TextVariants,
+  Grid,
+  GridItem
+} from "@patternfly/react-core";
+import {
+  ToggleOffIcon,
+  ToggleOnIcon,
+  CheckCircleIcon,
+  ErrorCircleOIcon
+} from "@patternfly/react-icons";
+import { InputText, JsonViewEditor } from "components";
+import { IInfoTypePlan } from "../GeneralInfo";
+
+export interface IMessagingObject {
+  url?: string;
+  username?: string;
+  password?: string;
+  addressSpace?: string;
+  eventsAddressName?: IInfoTypePlan;
+  telemetryAddressName?: IInfoTypePlan;
+  commandAddressName?: IInfoTypePlan;
+}
+export interface IAdapterConfig {
+  url?: string;
+  host?: string;
+  port?: number;
+  tlsEnabled?: boolean;
+}
+export interface IAdapter {
+  type: "http" | "https" | "mqtt" | "amqp" | "amqps" | "coap";
+  value: IAdapterConfig;
+}
+
+export interface IAccessCredentialsProps {
+  tenantId: string;
+  messaging: IMessagingObject;
+  adapters: IAdapter[];
+}
+
+const getCombinedString = (a: string, b?: string) => {
+  let s: string = "";
+  s += a;
+  if (b !== undefined) {
+    s += ", ";
+    s += b;
+  }
+  return s;
+};
+const AccessCredentials: React.FunctionComponent<IAccessCredentialsProps> = ({
+  tenantId,
+  messaging,
+  adapters
+}) => {
+  const [isHidden, setIsHidden] = useState<boolean>(false);
+  const onToggle = () => {
+    setIsHidden(!isHidden);
+  };
+  const EditIcon = isHidden ? (
+    <ToggleOffIcon onClick={onToggle} size="lg" />
+  ) : (
+    <ToggleOnIcon onClick={onToggle} size="lg" />
+  );
+  const data = {
+    access_credentials: {
+      tenantId: tenantId,
+      messaging: messaging,
+      adapters: adapters
+    }
+  };
+
+  const IsEnabled = (value: boolean) => {
+    return value ? (
+      <>
+        <CheckCircleIcon color={"green"} /> &nbsp;Enabled
+      </>
+    ) : (
+      <>
+        <ErrorCircleOIcon color={"red"} /> Disabled
+      </>
+    );
+  };
+  const Messaging = messaging && (
```

Comment:
can wrap this code in function besides variable and use destructing for reading the values to avoid using messaging again.
i.e. const {url,username,password}=messaging;